engine/robot_logic/traversal.py

```diff
@@ -38,14 +40,10 @@ def traverse_standard(robot_state, unvisited_waypoints, allowed_dist_error, data
         # TODO: add obstacle avoidance support
         # TODO: add return when tank is full, etc
         # Removed function move to target heading since it doesn't really make sense
+        robot_state.goal_location = curr_waypoint
         move_to_target_node(robot_state, curr_waypoint, allowed_dist_error, database)
         unvisited_waypoints.popleft()
-        if robot_state.phase == Phase.AVOID_OBSTACLE: # TODO: Temp fix until goal loc reset
-            robot_state.phase == Phase.AVOID_OBSTACLE
-            phase_change(robot_state)
-            robot_state.goal_location = curr_waypoint
-            robot_state.prev_phase = Phase.TRAVERSE
-            return robot_state, unvisited_waypoints
+        
         # TODO: THIS ISNT CORRECT: NEED TO CHECK IF AVOID_OBSTACLE IN move_to_target_node or can also make PID traversal a separate thread and stop the thread when obstacle detected
 
     robot_state.phase = Phase.RETURN
```

Comment:
line 51, wouldn't unvisited_waypoints be none? Also do you remember why we needed this before?

Reply:
I see issue, move_to_target_node should return a bool. If true, continue, else if false, break while loop and do not set phase to return. Then in move_to_target_node, return false if phase = obstacle avoidance, otherwise return true